staging/src/k8s.io/api/certificates/v1/types.go

```diff
@@ -84,6 +84,11 @@ type CertificateSigningRequestSpec struct {
 	//  6. Whether or not requests for CA certificates are allowed.
 	SignerName string `json:"signerName" protobuf:"bytes,7,opt,name=signerName"`
 
+	// notAfterHint is a hint to the signer in regards to when the issued certificate should expire.
+	// The signer may or may not honor this field.  The well-known kubernetes signers will honor this field
+	// as long as the requested notAfter time is not later than the maximum notAfter time they will honor.
+	NotAfterHint metav1.Time `json:"notAfterHint,omitempty" protobuf:"bytes,8,opt,name=notAfterHint"`
```

Comment:
You have to make this a pointer for it to actually be omitted when empty. This will remove the need for the after_roundtrip changes